apps/aecore/lib/aecore/structures/account.ex

```diff
@@ -31,63 +34,25 @@ defmodule Aecore.Structures.Account do
              locked: []}
   end
 
-  @spec tx_in!(Account.t() | nil, SpendTx.t(), integer()) :: Account.t()
-  def tx_in!(nil, tx, block_height) do
-    tx_in!(empty(), tx, block_height)
-  end
-  def tx_in!(account, tx, block_height) do
-    if block_height <= tx.lock_time_block do
-      if tx.value < 0 do
-        throw {:error, "Can't lock a negative transaction"}
-      end
-      new_locked = account.locked ++ [%{amount: tx.value, block: tx.lock_time_block}]
-      %Account{account | locked: new_locked}
-    else
-      new_balance = account.balance + tx.value
-      if new_balance < 0 do
-        throw {:error, "Negative balance"}
-      end
-      %Account{account | balance: new_balance}
-    end
-  end
+  @spec update_locked(Account.t(), integer()) :: Account.t()
```

Comment:
I preferred having the `tx_in!` and `tx_out!` functions here, otherwise this file makes no sense and is a good abstraction to not have all code in the chain state calculation. Should be easy to adjust with your implementation.

Reply:
The tx_in and tx_out functionality is moved to it's corresponding transaction module SpendTx. In my opinion it's the most correct place for them. Each transaction module should handle it's work on the blockchain. The accounts module on the other hand is usefull for reasons of abstraction and even if at the moment it doesn't have much functionality that may change in the future.

Reply:
It should be in account, as it affects the accounts state, otherwise the lock functionality makes no sense here as well. In the account you can call the function in the corresponding transaction module .

Reply:
So you mean that from the SpendTx module where i need to make the tx_in and tx_out I'll call them from the Account module like so: `Account.tx_in!()` and `Account.tx_out!()`